controllers/control/cassandratask_controller.go

```diff
@@ -0,0 +1,559 @@
+			return ctrl.Result{Requeue: true}, nil // TODO Add some sane time here
+		}
+		// TODO There are other tasks running, are they allowing or forbiding the concurrent work?
+		for _, task := range activeTasks {
+			if cassTask.Spec.ConcurrencyPolicy == nil || *task.Spec.ConcurrencyPolicy == batchv1.ForbidConcurrent {
+				logger.V(1).Info("this job isn't allowed to run due to ConcurrencyPolicy restrictions", "activeTasks", len(activeTasks))
+				return ctrl.Result{Requeue: true}, nil // TODO Add some sane time here
+			}
+		}
+	}
+
+	var res ctrl.Result
+	completedCount := int32(0)
+
+	// TODO We only support a single job at this stage (helps dev work)
+	if len(cassTask.Spec.Jobs) > 1 {
+		return ctrl.Result{}, fmt.Errorf("only a single job can be defined in this version of cass-operator")
+	}
+
+	// TODO We need to verify that the cluster is actually healthy before we run these..
+	taskId := string(cassTask.UID)
+
+	for _, job := range cassTask.Spec.Jobs {
+		// TODO This should check the status of the job from Status, not reconciling it (we overwrite the jobId in those cases)
+		switch job.Command {
+		case "rebuild":
+			res, err = r.reconcileEveryPodTask(ctx, taskId, &dc, rebuild(job.Arguments))
+		case "cleanup":
+			res, err = r.reconcileEveryPodTask(ctx, taskId, &dc, cleanup())
+		case "decommission":
+			// res, err = r.reconcileEveryPodTask(ctx, taskId, &dc, decommission())
+		default:
+			err = fmt.Errorf("unknown job command: %s", job.Command)
+			return ctrl.Result{}, err
+		}
+
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if res.RequeueAfter > 0 {
+			// This job isn't complete yet or there's an error, do not continue
+			break
+		}
+		completedCount++
+	}
+
+	taskPatch := client.MergeFrom(cassTask.DeepCopy())
+
+	if cassTask.GetLabels() == nil {
+		cassTask.Labels = make(map[string]string)
+	}
+
+	if _, found := cassTask.GetLabels()[taskStatusLabel]; found {
+		if res.RequeueAfter == 0 && !res.Requeue {
+			// Job has been completed
+			cassTask.GetLabels()[taskStatusLabel] = completedTaskLabelValue
+			if err = r.Client.Patch(ctx, &cassTask, taskPatch); err != nil {
+				return res, err
+			}
+
+			cassTask.Status.Active = 0
+			cassTask.Status.CompletionTime = &timeNow
+
+			// Requeue for deletion later
+			deletionTime := calculateDeletionTime()
+			nextRunTime := deletionTime.Sub(timeNow.Time)
+			res.RequeueAfter = nextRunTime
+			logger.V(1).Info("This task has completed, scheduling for deletion in " + nextRunTime.String())
+		}
+	} else {
+		// Starting the run, set the Active label so we can quickly fetch the active ones
+		cassTask.GetLabels()[taskStatusLabel] = activeTaskLabelValue
+
+		if err = r.Client.Patch(ctx, &cassTask, taskPatch); err != nil {
+			return res, err
+		}
+
+		cassTask.Status.StartTime = &timeNow
+		cassTask.Status.Active = 1 // We don't have concurrency inside a task at the moment
+	}
+
+	if cassTask.Status.Succeeded != completedCount {
+		cassTask.Status.Succeeded = completedCount
```

Comment:
`.Status.Succeeded` isn't getting updated properly. I am testing with 2 C* nodes and it has a value of 1 when the tasks finish.

Reply:
It's not a pod count, it's a job count inside the task (increased after a job has finished).